CustomEvents.cpp

```diff
@@ -5086,9 +5545,134 @@ HOOK_METHOD(CreditScreen, Done, () -> bool)
+            pausing -= trueScrollSpeed;
+        }
+        else if (std::max(scroll, -pausePosition) ==  -pausePosition && pausing > -pauseDuration) {
+            pausing -= trueScrollSpeed;
+        }
+        else {
+            scroll -= trueScrollSpeed;
+        }
+
+        /* ----- Rendering -----*/
+        // Background rendering
+        G_->GetResources()->RenderImage(bg, 0, 0, 0, colorGray, 1.f, false);
+        CSurface::GL_SetColor(colorWhite);
+        // Hardcoded Texts
+        if (!shipName.empty()) {
+            int yPos = static_cast<int>(creditTextStartHeight + scroll);
+            if (yPos > -100 && yPos < 800) {
+                freetype::easy_printNewlinesCentered(14, creditTextCentered, yPos, 1200, creditTextThankShip);
+            }
+            yPos += 40;
+            if (yPos > -100 && yPos < 800) {
+                freetype::easy_printNewlinesCentered(20, creditTextCentered, yPos, 1200, shipName);
+            }
+            if (!crewString.empty()) {
+                yPos += 80;
+                if (yPos > -720 && yPos < 800) {
+                    freetype::easy_printNewlinesCentered(14, creditTextCentered, yPos, 1200, creditTextThankCrew);
+                }
+                yPos += 40;
+                if (yPos > -crewStringCutOff && yPos < 800) {
+                    freetype::easy_printNewlinesCentered(18, creditTextCentered, yPos, 720, crewString);
+                }
+            } else {
+                yPos += 80;
+                if (yPos > -100 && yPos < 800) {
+                    freetype::easy_printNewlinesCentered(14, creditTextCentered, yPos, 1200, creditTextThankAi);
+                }
+            }
+            yPos += 246;
+            if (yPos > -100 && yPos < 800) {
+                freetype::easy_printNewlinesCentered(14, creditTextCentered, yPos, 750, creditTextVictory);
+            }
+        }
+        // xml Texts
+        for (const auto& creditText : creditTextValues) {
+            if ((currentHeight + scroll) > -creditText.cutOff && (currentHeight + scroll) < 800) {
+                freetype::easy_printNewlinesCentered(creditText.font, creditText.horizontal, static_cast<int>(currentHeight + scroll), creditText.lineLenght, G_->GetTextLibrary()->GetText(creditText.text, G_->GetTextLibrary()->currentLanguage));
+            }
+            currentHeight += creditText.spacing;
+        }
+        // credit.txt Names
+        int nameCount = 0;
+        for (const auto& creditFile : creditFileNames) {
+            if (nameCount == 0) {
+                creditNamesHorizontal = creditTextLeft;
+            } else {
+                creditNamesHorizontal = (nameCount == 1) ? creditTextCentered : creditTextRight;
+            }
+            if ((currentHeight + scroll) > -creditFile.cutOff && (currentHeight + scroll) < 800) {
+                freetype::easy_printNewlinesCentered(creditNamesFontSize, creditNamesHorizontal, static_cast<int>(currentHeight + scroll), 750, creditFile.names);
```

Comment:
This is what's causing the bug I've been pestering you about in the MV credits. The argument you're passing in for `line_length` is a static value `750`, which is not enough space for the text, so it automatically adds a newline when it renders it.